istioctl/pkg/validate/validate.go

```diff
@@ -300,9 +308,17 @@ func validateFiles(istioNamespace *string, defaultNamespace string, filenames []
 			if err != nil {
 				return err
 			}
-			if !info.IsDir() && filepath.Ext(path) == ".yaml" {
+
+			if info.IsDir() {
+				return nil
+			}
+
+			if isFileFormatValid(path) {
 				processFile(path)
+			} else {
+				log.Warnf("skipping file %v, recognized file extensions are: %v", path, fileExtensions)
```

Comment:
This is a bit of a changed behavior because I'm not sure we should warn about irrelevant files (like before)

Reply:
I think we don't need this warning, as we only validate resources that we care about. If we warn about every unrecognized file, we may end up with many spam logs when validating the directory.

Reply:
so should I just skip them silently?

Reply:
the reason why I marked it as warn is that there is similar part of the code in `istioctl analyze` which throws message to stderr for irrelevant files https://github.com/istio/istio/blob/dfcf08db3fd306335771b5fe1b1f3038d85dec3f/istioctl/pkg/analyze/analyze.go#L400

Reply:
I think it's fine to remove the warning since `analyze` provides more exhaustive info, even for like parsing errors, and here we are just validating Istio configs.